src/aof.c

```diff
@@ -928,6 +1362,108 @@ int loadAppendOnlyFile(char *filename) {
 cleanup:
     if (fakeClient) freeClient(fakeClient);
     fclose(fp);
+    return ret;
+}
+
+/* Load the AOF files according the aofManifest pointed by am. */
+int loadAppendOnlyFiles(aofManifest *am) {
+    serverAssert(am != NULL);
+    int ret = C_OK;
+    long long start;
+    off_t total_size = 0;
+    sds aof_name;
+
+    /* If there is no information about BASE AOF and INCR AOF in aofManifest, then 
+     * there is only one possibility: the aof manifest file does not exist, we may 
+     * be starting from an old redis version. so we must fall back to the previous 
+     * loading mode.
+     */
+    if (am->base_aof_info == NULL && listLength(am->incr_aof_list) == 0) {
+        struct redis_stat sb;
+        /* If the `server.aof_filename` file does not exist, we will directly return 
+         * AOF_NOT_EXIST, and redis will ignore this error. */
+        if (redis_stat(server.aof_filename, &sb) != 0) {
+            return AOF_NOT_EXIST;
+        }
+
+        /* If the server.aof_filename file exists, we manually construct a BASE 
+         * type aofInfo and add it to aofManifest. In this way, we can reuse the 
+         * following code to load this AOF file. */
+        aofInfo *ai = aofInfoCreate();
+        ai->file_name = sdsnew(server.aof_filename);
+        ai->file_seq = 1;
+        ai->file_type = AOF_FILE_TYPE_BASE;
+        am->base_aof_info = ai;
+        am->curr_base_aof_seq = 1;
+        am->dirty = 1;
+
+        int ret = persistAofManifest(am);
+        if (ret == C_ERR) {
+            return AOF_FAILED;
+        }
+    }
+
+    serverAssert(am->base_aof_info != NULL || listLength(am->incr_aof_list) != 0);
+
+    /* Here we calculate the total size of all BASE AOF and INCR AOF files in 
+     * advance, it will be set to `server.loading_total_bytes`. */
+    total_size = getBaseAndIncrAppendOnlyFilesSize(server.aof_manifest);
+    startLoading(total_size, RDBFLAGS_AOF_PREAMBLE, 0);
+
```

Comment:
@oranagra  PLZ review this part again, thanks.

Reply:
LGTM